sdk/dotnet/Pulumi/Deployment/DeploymentInstance.cs

```diff
@@ -49,6 +49,25 @@ public Task<T> InvokeAsync<T>(string token, InvokeArgs args, InvokeOptions? opti
         public Task InvokeAsync(string token, InvokeArgs args, InvokeOptions? options = null)
             => _deployment.InvokeAsync(token, args, options);
 
+        /// <summary>
+        /// Dynamically invokes the function '<paramref name="token"/>', which is offered by a
+        /// provider plugin.
+        /// <para/>
+        /// The result of <see cref="Call{T}"/> will be a <see cref="Output{T}"/> resolved to the
+        /// result value of the provider plugin.
+        /// <para/>
+        /// The <paramref name="args"/> inputs can be a bag of computed values(including, `T`s,
+        /// <see cref="Task{TResult}"/>s, <see cref="Output{T}"/>s etc.).
+        /// </summary>
+        public Output<T> Call<T>(string token, CallArgs args, Resource? self = null, CallOptions? options = null)
```

Comment:
When `self` is `null`, we don't pass along a `__self__` arg to the `Call` RPC and don't use it to determine the provider (if you want to explicitly specify the provider, you'd have to do it via `CallOptions`).

This is groundwork to enable module-level functions that use `Call` instead of `Invoke`, to allow module-level functions to accept `Input<T>`. We've deferred this for now, but eventually we want to have a way in the schema to opt-in to this for module-level functions so they can accept `Input<T>`, and these will use `Call` without a `__self__` arg, rather than `Invoke`.

The reason for the new `Call` RPC (rather than reusing `Invoke`) is to make it easier for provider authors to implement `Call` -- we provide helpers that adapt the raw RPC values into the Pulumi SDK model of each language (e.g. the values are converted back into `Output<T>` as needed) in the provider.